qiskit/providers/ibmq/jupyter/dashboard/dashboard.py

```diff
@@ -154,90 +231,127 @@ def stop_dashboard(self) -> None:
         """
-        do_pop = False
-        ind = None
-        for idx, job in enumerate(self.jobs):
-            if job.job_id == job_id:
-                do_pop = True
-                ind = idx
-                break
-        if not do_pop:
+        # Establish which jobs widget list to use (Runtime or Circuit)
+        jobs_wlist = self.rt_jobs if job_type == JobType.Runtime else self.jobs
+        # Find job widget with correct `job_id`
+        jobw = next((job for job in jobs_wlist if getattr(job, 'job_id') == job_id), None)
+        if jobw is None:
             raise Exception('Job is not found.')
-        if self.jobs[ind].children[3].value not in ['CANCELLED',
-                                                    'DONE',
-                                                    'ERROR']:
+        # If the job status is not active, Cancel.
+        if jobw.status not in ['CANCELLED', 'DONE', 'ERROR']:
             try:
-                self.jobs[ind].job.cancel()
-                status = self.jobs[ind].job.status()
+                jobw.job.cancel()
+                status = jobw.job.status()
             except IBMQJobApiError:
                 pass
             else:
-                self.update_single_job((self.jobs[ind].job_id,
-                                        status.name, 0,
-                                        status.value))
+                self.update_single_job(job_type, (job_id, status.name, 0, status.value))
+
+    def clear_done(self, job_type: JobType) -> None:
+        """Clear the done jobs from the list.
 
-    def clear_done(self) -> None:
-        """Clear the done jobs from the list."""
-        _temp_jobs = []
+        Args:
+            job_type: the job type
+        """
+        # Determine the list of job widgets to use
+        jobs = self.jobs if job_type == JobType.Circuit else self.rt_jobs
+
+        active_jobs = []
         do_refresh = False
-        for job in self.jobs:
-            job_str = job.children[3].value
-            if not (('DONE' in job_str) or ('CANCELLED' in job_str) or ('ERROR' in job_str)):
-                _temp_jobs.append(job)
+        # Find the active jobs
+        for job in jobs:
+            if job.status not in ['DONE', 'CANCELLED', 'ERROR']:
+                active_jobs.append(job)
             else:
+                # Close the job and ensure the UI is refreshed
                 job.close()
                 do_refresh = True
+        # Refresh the UI with the active jobs
         if do_refresh:
-            self.jobs = _temp_jobs
-            self.refresh_jobs_board()
+            if job_type == JobType.Circuit:
+                self.jobs = active_jobs
+            else:
+                self.rt_jobs = active_jobs
+        self.refresh_jobs_board(job_type)
 
     def _init_subscriber(self) -> None:
         """Initializes a subscriber that listens to job start events."""
 
-        def _add_job(job: IBMQJob) -> None:
+        def _add_job(job: Job) -> None:
```

Comment:
nit - `RuntimeJob` is not of type `Job`, so this technically is `Union[IBMQJob, RuntimeJob]`.

Reply:
In the great `git force push` crash of 2021, `RuntimeJob` being made a subclass of `Job` was lost. This is updated now!

It had almost identical methods to `Job`, so making it a subclass actually reduced the size of the code (2 functions were redundant and were removed, 1 was added but given a `NotImplemented` error if used)